monkeytype/compat.py

```diff
@@ -7,6 +7,14 @@
 
 from mypy_extensions import _TypedDictMeta  # type: ignore[attr-defined]
 
+try:
+    from django.utils.functional import cached_property as cp  # type: ignore[import]
+
+    cached_property = cp
+except ImportError:
+    # Django may not be installed
+    cached_property = None
```

Comment:
This would be more a new feature in addition, rather than a replacement. Django has not adopted the stdlib `cached_property` and lots of people I'm sure are still using Django's version, so we can't just drop support for it. (Note that MonkeyType's usage of `cached_property` is not "we need a cached_property implementation for our own use", it's "we have special support for profiling types returned from cached properties," so it's a feature more than a dependency.)

Reply:
Also sadly the stdlib `cached_property` (which I added to Python 3.8) is badly inefficient when used with threads, because it locks all instances of the cached property on update of any individual instance), which is why Django and many others have not adopted it, but that's a separate issue...